app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt

```diff
@@ -0,0 +1,161 @@
+package eu.kanade.tachiyomi.ui.setting.database
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.os.Bundle
+import android.view.*
+import androidx.core.view.*
+import androidx.core.view.WindowInsetsCompat.Type.systemBars
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import eu.davidea.flexibleadapter.FlexibleAdapter
+import eu.davidea.flexibleadapter.Payload
+import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.databinding.ClearDatabaseControllerBinding
+import eu.kanade.tachiyomi.ui.base.controller.DialogController
+import eu.kanade.tachiyomi.ui.base.controller.NucleusController
+import eu.kanade.tachiyomi.util.system.toast
+import eu.kanade.tachiyomi.util.view.activityBinding
+import eu.kanade.tachiyomi.util.view.fullAppBarHeight
+import eu.kanade.tachiyomi.util.view.liftAppbarWith
+import eu.kanade.tachiyomi.util.view.scrollViewWith
+
+class ClearDatabaseController :
+    NucleusController<ClearDatabaseControllerBinding, ClearDatabasePresenter>(),
+    FlexibleAdapter.OnItemClickListener {
+
+    private var recycler: RecyclerView? = null
+    private var adapter: FlexibleAdapter<ClearDatabaseSourceItem>? = null
+    private var menu: Menu? = null
```

Comment:
so this menu isn't used. Its assigned but never used.

Reply:
so it looks like this menu is used in https://github.com/Jays2Kings/tachiyomiJ2K/blob/9b17f8b9c0a97e5fbb3c3a6552fcb4a99456f5df/app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt#L86-L88

Reply:
yeah sorry for the confusion lol im 100% on this now, just replacing invalidate options menu with menu?.forEach { menuItem -> menuItem.isVisible = size > 0 }

Reply:
to confirm, you're saying replace https://github.com/Jays2Kings/tachiyomiJ2K/blob/9b17f8b9c0a97e5fbb3c3a6552fcb4a99456f5df/app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt#L118
with `menu?.forEach { menuItem -> menuItem.isVisible = size > 0 }` right?

Reply:
yup